R/template.R

```diff
@@ -1,110 +1,184 @@
-#' it from being passed on to the metric function implementation. For
-#' classification metrics, this can either be `NULL` to use the default
-#' `event_level` value of the `metric_fn` or a single string of either
-#' `"first"` or `"second"` to pass along describing which level should be
-#' considered the "event".
+#' @param event_level This can either be `NULL` to use the default `event_level`
+#' value of the `fn` or a single string of either `"first"` or `"second"`
+#' to pass along describing which level should be considered the `"event"`.
 #'
 #' @param case_weights For metrics supporting case weights, an unquoted
 #' column name corresponding to case weights can be passed here. If not `NULL`,
-#' the case weights will be passed on to `metric_fn` as the named argument
+#' the case weights will be passed on to `fn` as the named argument
 #' `case_weights`.
 #'
-#' @param ... Currently not used. Metric specific options are passed in
-#' through `metric_fn_options`.
 #'
-#' @param metric_fn_options A named list of metric specific options. These
+#' @param fn_options A named list of metric specific options. These
 #' are spliced into the metric function call using `!!!` from `rlang`. The
 #' default results in nothing being spliced into the call.
 #'
-#' @seealso [metric_vec_template()] [finalize_estimator()] [dots_to_estimate()]
-#'
+#' @keywords internal
 #' @export
-metric_summarizer <- function(metric_nm,
-                              metric_fn,
-                              data,
-                              truth,
-                              estimate,
-                              estimator = NULL,
-                              na_rm = TRUE,
-                              event_level = NULL,
-                              case_weights = NULL,
-                              ...,
-                              metric_fn_options = list()) {
+class_metric_summarizer <- function(name,
+                                    fn,
+                                    data,
+                                    truth,
+                                    estimate,
+                                    ...,
+                                    estimator = NULL,
+                                    na_rm = TRUE,
+                                    event_level = NULL,
+                                    case_weights = NULL,
+                                    fn_options = list()) {
+  rlang::check_dots_empty()
+
   truth <- enquo(truth)
   estimate <- enquo(estimate)
   case_weights <- enquo(case_weights)
 
-  validate_not_missing(truth, "truth")
-  validate_not_missing(estimate, "estimate")
+  # Explicit handling of length 1 character vectors as column names
+  nms <- colnames(data)
+  truth <- handle_chr_names(truth, nms)
+  estimate <- handle_chr_names(estimate, nms)
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = name,
+    .estimator = finalize_estimator(!! truth, estimator, name),
```

Comment:
I think the rlang style is to _not_ put a space between `!!` and the name, so `!!truth`. I don't think we were sure about this when I was working on this function originally but it is definitely the standard now
https://rlang.r-lib.org/reference/injection-operator.html#injecting-expressions

Reply:
done in https://github.com/tidymodels/yardstick/pull/322/commits/6d7bea2aad99dcd0b07a312cf464c0b26561ea89